### PR TITLE
Service dtos, services and controller (all tests pass)

### LINKED ISSUE #65: Implementation and Testing of Service class
- [x] Create a new branch for Service class 
- [x] Change from "to do" to "in progress"
- [x] Create service skeleton 
- [x] Create controller skeleton
- [x] Create DTO skeleton
- [x] Create test class called TestServiceService under src/test/java ca.mcgill.ecse321.repairshop.service package
- [x] Write Service tests

When a task is done: 💯 
_Don't forget to check the checkboxes + add a comment before closing an issue_
_Feel free to create more tasks_

**Links:**
[Tutorial notes](https://mcgill-ecse321-fall2020.github.io/ecse321-tutorial-notes/#_creating_restful_web_services_in_spring)
[Tutorial 6 - REST API](https://lrs.mcgill.ca/embed?rid=6b0fbc0e-223f-4599-9478-bcff69d42270&d2lhost=mycourses2.mcgill.ca)

### PR DESCRIPTION
**Pull request by T-Syme**

### CODE REVIEW

**Review by Shaswata79 — APPROVED** [review]

**Review by JDziewonski98 — APPROVED** [review]
LGTM

File: RepairShop-Backend/src/test/java/ca/mcgill/ecse321/repairshop/service/TestServiceService.java [hunk 1]
```diff
@@ -1,4 +1,120 @@
...
+
+        try {
+            serviceDto = serviceService.createService(null, 0, 0);
+        } catch (Exception e) {
+            error = e.getMessage();
+        }
+
+        assertNull(serviceDto);
+        assertEquals("A service name is required", error);
+
```
Comment (sandyl289) [inline RepairShop-Backend/src/test/java/ca/mcgill/ecse321/repairshop/service/TestServiceService.java:113]:
```suggestion
    @Test
    public void testGetAllServices() {

        List<ServiceDto> serviceDtos = serviceService.getAllServices();
        assertEquals(1, serviceDtos.size()); // List contains a single element.
        ServiceDto serviceDto = (ServiceDto) serviceDtos.toArray()[0];
        assertEquals(SERVICE_NAME, serviceDto.getName()); // List contains the only comment.
        assertEquals(SERVICE_DURATION, serviceDto.getDuration()); // List contains the only comment.
        assertEquals(SERVICE_PRICE, serviceDto.getPrice()); // List contains the only comment.
        
    }
```

File: RepairShop-Backend/src/test/java/ca/mcgill/ecse321/repairshop/service/TestServiceService.java [hunk 2]
```diff
@@ -1,4 +1,120 @@
...
+    }
```
Comment (sandyl289) [inline RepairShop-Backend/src/test/java/ca/mcgill/ecse321/repairshop/service/TestServiceService.java:114]:
```suggestion
    @Test //valid service name
    public void testServiceByName() {
        ServiceDto serviceDto = null;
        try{
            serviceDto = serviceService.getServiceByName(SERVICE_NAME);

        } catch (Exception e) {
            fail(e.getMessage());
        }
        
        assertNotNull(serviceDto);
        assertEquals(SERVICE_NAME, serviceDto.getName());
        assertEquals(SERVICE_DURATION, serviceDto.getDuration());
        assertEquals(SERVICE_PRICE, serviceDto.getPrice());
    }
    
        @Test //invalid service name
    public void testServiceByNameNonExistent() {
        String error=null;
        ServiceDto serviceDto = null;
        String name = "I don't exist";
        try{
            serviceDto = serviceService.getServiceByName(name);

        } catch (Exception e) {
            error = e.getMessage();
        }

        assertNull(serviceDto);
        assertEquals("Could not find service with name "+name, error);
    }
    }
```

**Review by sandyl289 — CHANGES_REQUESTED** [review]
I added some  tests suggestion  for getServiceByName and getAllServices().
Other than that, I think we are ready to merge it to the main branch! 💯

**Comment by sandyl289** [comment]
I'm not sure if we should link this pull request to the [Add service (Admin)](https://github.com/McGill-ECSE321-Winter2021/project-group-09/issues/94), since closing this pull request will also close that issue. 
The "Add service (Admin)"  issue is not complete yet, since we'll need to verify that the user logged in is the Admin.

image

**Comment by T-Syme** [comment]
> I'm not sure if we should link this pull request to the [Add service (Admin)](https://github.com/McGill-ECSE321-Winter2021/project-group-09/issues/94), since closing this pull request will also close that issue.
> The "Add service (Admin)" issue is not complete yet, since we'll need to verify that the user logged in is the Admin.

You're right, thanks!

**Review by sandyl289 — APPROVED** [review]
I think we can merge it to master now

**Pull request merged**